### PR TITLE
Attention mask with cuda graph

### PR DESCRIPTION
**Pull request by wangyems**
To keep attention_mask as model's input when using cuda graph, we assume shared_kv_cache is used and attention_mask is allocated with [max_batch, max_seq_length]

1. revert seqlens_k and total_seq_len in model builder and GenAI tool.
2. allocate attention_mask with fixed size [max_batch, max_seq_length] if using cuda graph
3. add readme

### CODE REVIEW

File: src/python/py/models/README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 This folder contains the model builder for quickly creating optimized and quantized ONNX models within a few minutes that run with ONNX Runtime GenAI.
 
 # Contents
+
```
Comment (kunal-vaishnavi) [inline src/python/py/models/README.md:6]: Can you add `Enable CUDA Graph` as a section in the table of contents?

File: src/models/kernels.cu [hunk 2]
```diff
@@ -1,71 +1,87 @@
...
-    } else {
-      mask_data[i * current_length + j] = 1;
+__global__ void CopyAndUpdateAttentionMask(T *mask_data, const T *old_mask_data, int batch_beam_size,
+                                           int current_length, int max_length) {
+    int global_index = blockIdx.x * blockDim.x + threadIdx.x;
+    int i = global_index / current_length;
+    int j = global_index % current_length;
+    if (i < batch_beam_size) {
+        if (j < current_length - 1) {
+            mask_data[i * max_length + j] = old_mask_data[i * (current_length - 1) + j];
```
Comment (yufenglee) [inline src/models/kernels.cu:29]: >max_length [](http://example.com/codeflow?start=26&length=10)

is it max_length or current_length? Do you need a max_length for copy and update scenario?

**Review by yufenglee — APPROVED** [review]
:shipit:

File: src/models/kernels.cu (hunk 2 above)
Comment (wangyems) [reply src/models/kernels.cu:29]: The new mask uses max length and the old mask uses current length because we allocate max length to mask and enable cuda graph in the second iteration

**Review by kunal-vaishnavi — APPROVED** [review]
